### ISSUE TITLE
Vend official React wrappers

### ISSUE THREAD

**Issue by e111077** [material-components/material-web#4432]
Use `@lit-labs/react` to generate react wrappers for our components

**Comment by MichaelPriebe** [comment]
How about https://github.com/BuilderIO/mitosis so every framework can use the components?

**Comment by e111077** [comment]
Several reasons as to why we are using Lit:

- web components work everywhere. This bug is just for a more idiomatic feel for React users
- we have first party clients at Google so we are subject to Google's infrastructure
   - i.e. the heavy use of Sass, and mitosis is not allowed internally
- having the Lit team at Google means that we can get personalized support for the underlying library
- web components serve the same use case as mitosis without additional build complexity

**Comment by e111077** [comment]
**UPDATE:** probably going to push this post 1.0 but offer documentation on how to make your own with `@lit-labs/react`

**Comment by Stefanyshyn** [comment]
@e111077, @asyncLiz  Yo there!
Thanks for recommending to use of the `@lit-labs/react` package to display components in react. It was very helpful. I appreciate that.

I also use it to render the icon component. I have a problem rendering this component. This problem occurs if I want to pass the `&#xe834` Unicode code point to the component. [This link lists the valid values for the icon component](https://github.com/material-components/material-web/blob/main/docs/components/icon.md#usage).

Could you help me find out what the problem is and how to fix it? Any suggestions are welcome

[Here is a link to the problem where you can look at it in more detail](https://codesandbox.io/s/material-web-react-p9frts)

**Comment by e111077** [comment]
Hello, I believe there is a typo in that example. Valid Unicode should end in a semicolon `;` in HTML. 

https://codesandbox.io/s/material-web-react-forked-kkyfwl

**Comment by EWhite613** [comment]
@e111077 would there be a standard you have for events naming?

Like[ Lit docs](https://lit.dev/docs/frameworks/react/) always have `on<event>: 'event'`, I assume this would follow?


Would be nice to know so once we get an official one, I could hot swap my code to the official react wrappers without much migration needed.

**Comment by e111077** [comment]
Yes, we would likely do a direct camel case event name conversion e.g.

`close-menu` -> `onCloseMenu`
`input` -> `onInput` (rather than `onChange` unless that _really_ breaks things in forms)

**Comment by SaadBazaz** [comment]
We ([Grayhat](https://grayhat.studio)) are working on a thin React wrapper which is in-sync with this repository. It's in its early stages, we'd appreciate any help.

https://github.com/grayhatdevelopers/material-web-react
